File: Real-Time-Search-Workshop/javascript/code/all_labs.js
/**
 * @fileoverview Executes all labs
 * @maker Joey Whelan
 */

import { Lab1 } from "./lab1.js";
import { Lab2 } from "./lab2.js";
import { Lab3 } from "./lab3.js";
import { Lab4 } from "./lab4.js";
import { Lab5 } from "./lab5.js";
import { Lab6 } from "./lab6.js";

(async () => {
    const lab1 = new Lab1();
    const client = await lab1.run();

    const lab2 = new Lab2();
    await lab2.run(client);

    const lab3 = new Lab3();
    await lab3.run(client);

    const lab4 = new Lab4();
    await lab4.run(client);

    const lab5 = new Lab5();
    await lab5.run(client);
    
    const lab6 = new Lab6();
    await lab6.run(client);

    await client.disconnect();
})();

File: Real-Time-Query-Workshop/dotnet/code/allLabs.cs
/**
 * @fileoverview Executes all labs
 * @maker Joey Whelan
 */

using SearchWorkshop;
using StackExchange.Redis;

public class AllLabs
{
    static void Main(string[] args)
    {
        Lab1 lab1 = new Lab1();
        IDatabase db = lab1.Run();

        Lab2 lab2 = new Lab2();
        lab2.Run(db);

        Lab3 lab3 = new Lab3();
        lab3.Run(db);

        Lab4 lab4 = new Lab4();
        lab4.Run(db);

        Lab5 lab5 = new Lab5();
        lab5.Run(db);

        Lab6 lab6 = new Lab6();
        lab6.Run(db);
    }
}